Remove advanced settings dependency for IMAP (#45184)
home-assistant/home-assistant.github.io: source/_integrations/imap.markdown

Some IMAP services, like Yahoo, require a `US-ASCII` charset to be configured.

### Selecting message data to include in the IMAP event

> By default, the IMAP event won't include `text` or `headers` message data. If you want them to be included (`text` or `headers`, or both), you have to manually select them in the option flow.

Reviewer: ```suggestion
By default, the IMAP event won't include `text` or `headers` message data. If you want them to be included (`text`, `headers`, or both), you have to manually select them in the option flow.
```

folder:
  description: The IMAP folder configuration
text:

>   description: The email body `text` of the message. By default, only the first 2048 bytes of the body text will be available, the rest will be clipped off. You can increase the maximum text size of the body, but this is not advised and will never guarantee that the whole message text is available. A better practice is using a custom event data template that can be used to parse the whole message, not limited by size. The rendered result will then be added as attribute `custom` to the event data to be used for automations. `text` will be included if it is explicitly selected in the option flow.

Reviewer: ```suggestion
  description: The email body `text` of the message. By default, only the first 2048 bytes of the body text will be available; the rest will be clipped off. You can increase the maximum text size of the body, but this is not advised and will never guarantee that the entire message text is available. A better practice is to use a custom event data template that can parse the entire message, not limited by size. The rendered result will then be added as an attribute `custom` to the event data to be used for automations. `text` will be included if it is explicitly selected in the option flow.
```

The `event_type` for the custom event should be set to `imap_content`. The configuration below shows how you can use the event data in a template `sensor`.

> If the default maximum message size (2048 bytes) to be used in events is too small for your needs, then this maximum size setting can be increased.

Reviewer: ```suggestion
If the default maximum message size (2048 bytes) used in events is too small for your needs, you can increase this maximum size.
```
